### PR TITLE
fix: Navigation Settings API

### LINKED ISSUE #19163: [Task]: APIs for published and unpublished Navigation config
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

- Adding new database schemas 
- Update existing API for GET /pages (with added nav schema)
- Update existing API for PUT /applications/{applicationId} (which can update nav schema)
- Test Cases

### PR DESCRIPTION
**Pull request by NilanshBansal**
## Description
This PR bug fixes the API for updating Navigation Settings (implemented by #19272).

Fixes #19163 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Manual

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/domains/Application.java [hunk 1]
```diff
@@ -110,8 +110,10 @@ public class Application extends BaseDomain {
 
     EmbedSetting embedSetting;
 
+    @JsonIgnore
     NavigationSetting unpublishedNavigationSetting;
 
+    @JsonIgnore
     NavigationSetting publishedNavigationSetting;
```
Comment (vishnu-gp) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/domains/Application.java:117]: @NilanshBansal Aren't we gonna export the nav settings part of the app export? Would that work with JsonIgnore?

File: app/server/appsmith-server/src/main/java/com/appsmith/server/domains/Application.java [hunk 2]
```diff
@@ -110,8 +110,10 @@ public class Application extends BaseDomain {
...
```
Comment (vishnu-gp) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/domains/Application.java:118]: @NilanshBansal Also, can you add a test case to verify your fix ?

File: app/server/appsmith-server/src/main/java/com/appsmith/server/domains/Application.java (hunk 1 above)
Comment (NilanshBansal) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/domains/Application.java:117]: @vishnu-gp Yes, app export works fine with JsonIgnore.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/domains/Application.java (hunk 2 above)
Comment (NilanshBansal) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/domains/Application.java:118]: Added test cases @vishnu-gp

**Review by vishnu-gp — DISMISSED** [review]

**Review by vishnu-gp — APPROVED** [review]

**Pull request merged**